### ISSUE TITLE
How is the eq (4) in the paper implemented? 

### ISSUE THREAD

**Issue by abhishekaich27** [Muzammal-Naseer/CDA#6]
It's not directly clear how you have implemented eq (4) in `L143--L145` in `train.py`. Some quick inputs?

**Comment by Muzammal-Naseer** [comment]
Hi, 

Equation (4) represents loss objective for the targetted attack. y and y' are the original and target labels respectively. similarly, x and x' are the original and adversarial images, respectively.

**Comment by abhishekaich27** [comment]
Just to confirm once more, won't the implementation in L145 in train.py, then say
loss = criterion(model(normalize(adv)), label) + criterion(model(normalize(img)), gt_label) ?

**Comment by Muzammal-Naseer** [comment]
loss = criterion(model(normalize(adv)), targte_label) + criterion(model(normalize(img)), original_label)

**Comment by abhishekaich27** [comment]
Correct, but the current L145 in train.py says just 
loss = criterion(model(normalize(adv)), targte_label)!

**Comment by Muzammal-Naseer** [comment]
Thanks for pointing that out...I will update it
